frontend/src/components/ConnectionPipe.js

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import * as tools from "../functions";
+import { map } from "../fakeData";
+
+class ConnectionPipe extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  componentDidUpdate() {}
+
+  render() {
+    // calculate city position
+    var city1 = map.cities[this.props.startCity];
```

Comment:
nit: I believe `var` is obsolete. Use `let` instead. Or, use `const` in cases like this where you don't need to reassign the value to make your code a bit more self-documenting.

Reply:
Fixed throughout.